model/src/lib.rs

```diff
@@ -45,41 +45,49 @@ pub struct UserOrder {
     pub valid_to: u32,
     pub app_data: u32,
     #[serde(with = "u256_decimal")]
-    pub tip: U256,
+    pub fee_amount: U256,
     pub order_kind: OrderKind,
     pub partially_fillable: bool,
     pub signature: Signature,
 }
 
-impl UserOrder {
+impl OrderCreation {
     pub fn token_pair(&self) -> Option<TokenPair> {
         TokenPair::new(self.buy_token, self.sell_token)
     }
 }
 
-/// An order that is returned when querying the orderbook.
-///
-/// Contains extra fields thats are populated by the orderbook.
-#[derive(Eq, PartialEq, Clone, Copy, Debug, Deserialize, Serialize)]
+/// An order as provided to the orderbook by the frontend.
+#[derive(Eq, PartialEq, Clone, Debug, Deserialize, Serialize)]
 #[serde(rename_all = "camelCase")]
-pub struct Order {
-    pub creation_time: DateTime<Utc>,
+pub struct OrderMetaData {
+    pub creation_date: DateTime<Utc>,
     #[serde(with = "h160_hexadecimal")]
     pub owner: H160,
-    #[serde(flatten)]
-    pub user_provided: UserOrder,
+    pub uid: String,
 }
 
-impl Default for Order {
+impl Default for OrderMetaData {
     fn default() -> Self {
+        let owner: H160 = Default::default();
         Self {
-            creation_time: DateTime::from_utc(NaiveDateTime::from_timestamp(0, 0), Utc),
-            owner: Default::default(),
-            user_provided: Default::default(),
+            creation_date: DateTime::from_utc(NaiveDateTime::from_timestamp(0, 0), Utc),
+            owner: owner,
+            uid: String::from("owner"),
         }
     }
 }
 
+/// An order that is returned when querying the orderbook.
+///
+/// Contains extra fields thats are populated by the orderbook.
+#[derive(Eq, PartialEq, Clone, Default, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Order {
+    pub order_meta_data: OrderMetaData,
+    pub order_creation: OrderCreation,
+}
```

Comment:
I don't think this matches the openapi spec because there is no `#[serde(flatten)]` anymore which is why you had to change the test to use individual structs.

Reply:
Do you think it should be returned flattened?

The openapi  PR does not mention the fattening anymore.

Reply:
I think it does because it uses

```
Order:
      allOf:
        - $ref: "#/components/schemas/OrderCreation"
        - $ref: "#/components/schemas/OrderMetadata"
```

we should load it in https://editor.swagger.io/ to confirm. I just tried this but it looks like there are some errors in the spec that the editor complains about.